doc(*): Backport deprecated Kubernetes API doc

hickeyma opened the pull request:
Back port of https://github.com/helm/helm-www/pull/559

Partial-Fix #7219 

**What this PR does / why we need it**:
This documentation helps users when Kubernetes APIs are deprecated in a Kubernetes version and the user is unable to upgrade a Helm release as a consequence.

**Special notes for your reviewer**:
The doc is largely based off the Helm v3 doc. The "Updating API Versions of a Release Manifest" section however is changed to represent the release object format in Helm v2.

Shoutout to @Arabus for the script he made available in https://github.com/helm/helm/issues/7219#issuecomment-568175917 for Helm v2.

**If applicable**:
- [x] this PR contains documentation

hickeyma committed:
Add deprecated Kubernetes API doc

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma commented:
@Arabus I would appreciate if you could take a look and see if it covers the areas for Helm v2?

Arabus commented:
Will do this evening after putting the little ones to sleep :-)

Arabus commented:
Ok I had a look and this looks very good already :+1: 

Two nitpicks: 
one should ensure to use the protobuf for the deployed tiller otherwise the decoding might fail
you can skip the use of cat when decoding because grep can open that file by itself ;-)

HTH

mattfarina reviewed (approved):
lgtm

hickeyma committed:
Update review comment

Review comment:
- https://github.com/helm/helm/pull/8135#issuecomment-629527051

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma commented:
Thanks for the review @Arabus. I added a note on using the correct `protobuf schema`. That is a good observation about using `grep` on the file directly. I am however going to leave for now so it is consistent with Helm 3 doc.

hickeyma commented:
Thanks for the review @mattfarina. I just added a note on `protobuf schema`.

mattfarina reviewed (approved):
lgtm

hickeyma committed:
Pull review comments from v3 doc

Pulled in review comments from
https://github.com/helm/helm-www/pull/639

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

mattfarina commented:
@hickeyma the reason the tests are failing is that google.golang.org/genproto used to be hosted at https://github.com/google/go-genproto. Google moved it to https://github.com/googleapis/go-genproto. The VCS location has changed which is causing problems because the cache knows the old location.

hickeyma commented:
Thanks for fixing the issue @mattfarina.

mattfarina reviewed (approved):
lgtm
